### PR TITLE
Fastboot: optionally disable VP8, VP9, HEVC decoders

### PR DESCRIPTION
**Pull request by IlyaSergeyev**
VP8, VP9, HEVC decoders flags moved into mfxconfig.h.in

### CODE REVIEW

File: builder/ConfTargets.cmake [hunk 1]
```diff
@@ -76,6 +76,7 @@ option( MFX_ENABLE_USER_DECODE "Enabled user decode plugins?" ON)
 option( MFX_ENABLE_USER_ENCODE "Enabled user encode plugins?" ON)
 option( MFX_ENABLE_USER_ENC "Enabled user ENC plugins?" ON)
 option( MFX_ENABLE_USER_VPP "Enabled user VPP plugins?" ON)
+option( MFX_ENABLE_VP9_VIDEO_DECODE "Enabled VP9 decoder?" ON)
```
Comment (dvrogozh) [inline builder/ConfTargets.cmake]: That's wrong. And means that you don't check what you have done. Options which you place _after_ include(${MFX_CONFIG_FILE}) won't take effect! I will submit patch to fix that for the bunch of USER options above.

File: mfxconfig.h.in [hunk 2]
```diff
@@ -34,3 +34,7 @@
 #if defined(MFX_ENABLE_VP8_VIDEO_DECODE) && defined(MFX_VA) &&!defined(AS_VP8D_PLUGIN)
 #define MFX_ENABLE_VP8_VIDEO_DECODE_HW
 #endif
+
+#if !defined(AS_HEVCD_PLUGIN)
```
Comment (dvrogozh) [inline mfxconfig.h.in]: I do not think that AS_HEVCD_PLUGIN is actually needed any more at the first place. Let's try to dispose of them as a first step. I think that the only one which may potentially be needed is AS_HEVCE_PLUGIN and maybe same for LA.

**Comment by IlyaSergeyev** [comment]
rebased

File: _studio/shared/include/mfx_config.h [hunk 3]
```diff
@@ -164,4 +160,12 @@
 	#define MFX_ENABLE_FOURCC_RGB565
 #endif
 
+#if defined(MFX_ENABLE_VP9_VIDEO_DECODE)
```
Comment (dvrogozh) [inline _studio/shared/include/mfx_config.h]: move this change higher right after `#if defined(MFX_ENABLE_H265_VIDEO_ENCODE)`, i.e. place similar things together

File: builder/ConfTargets.cmake [hunk 4]
```diff
@@ -80,6 +80,9 @@ option( MFX_ENABLE_USER_DECODE "Enabled user decode plugins?" ON)
...
```
Comment (dvrogozh) [inline builder/ConfTargets.cmake]: add empty line between MFX_ENABLE_USER_VPP and MFX_ENABLE_VP9_VIDEO_DECODE, please

File: mfxconfig.h.in [hunk 5]
```diff
@@ -29,3 +29,6 @@
 #cmakedefine MFX_ENABLE_H264_VIDEO_FEI_ENCODE
 #cmakedefine MFX_ENABLE_H265_VIDEO_ENCODE
 #cmakedefine MFX_ENABLE_HEVC_VIDEO_FEI_ENCODE
+#cmakedefine MFX_ENABLE_VP9_VIDEO_DECODE
```
Comment (dvrogozh) [inline mfxconfig.h.in]: VP8 should precede VP9

**Pull request merged**